src/CommunityToolkit.Aspire.Hosting.SurrealDb/SurrealDbBuilderExtensions.cs

```diff
@@ -0,0 +1,418 @@
+
+                return [
+                    new ContainerFile
+                    {
+                        Name = "instance.json",
+                        Contents = await WriteSurrealistInstanceJson(
+                            surrealDbServerInstances,
+                            surrealDbNamespaceResources, 
+                            surrealDbDatabaseResources, 
+                            cancellationToken
+                        ).ConfigureAwait(false),
+                    },
+                ];
+            });
+
+        configureContainer?.Invoke(surrealistContainerBuilder);
+
+        return builder;
+    }
+    
+    private static async Task<string> WriteSurrealistInstanceJson(
+        IList<SurrealDbServerResource> surrealDbServerInstances,
+        IList<SurrealDbNamespaceResource> surrealDbNamespaceResources,
+        IList<SurrealDbDatabaseResource> surrealDbDatabaseResources,
+        CancellationToken cancellationToken
+    )
+    {
+        using var stream = new MemoryStream();
+        await using var writer = new Utf8JsonWriter(stream, new JsonWriterOptions { Indented = true });
+
+        writer.WriteStartObject();
+
+        writer.WriteStartArray("connections");
+
+        foreach (var surrealInstance in surrealDbServerInstances)
+        {
+            if (surrealInstance.PrimaryEndpoint.IsAllocated)
+            {
+                SurrealDbNamespaceResource? uniqueNamespace = null;
+                SurrealDbDatabaseResource? uniqueDatabase = null;
+
+                var serverNamespaces = surrealDbNamespaceResources
+                    .Where(ns => ns.Parent == surrealInstance)
+                    .ToList();
+
+                if (serverNamespaces.Count == 1)
+                {
+                    uniqueNamespace = serverNamespaces.First();
+
+                    var nsDatabases = surrealDbDatabaseResources
+                        .Where(db => db.Parent == uniqueNamespace)
+                        .ToList();
+
+                    if (nsDatabases.Count == 1)
+                    {
+                        uniqueDatabase = nsDatabases.First();
+                    }
+                }
+
+                var endpoint = surrealInstance.PrimaryEndpoint;
+
+                writer.WriteStartObject();
+
+                writer.WriteString("id", surrealInstance.Name);
+                writer.WriteString("name", surrealInstance.Name);
+
+                if (uniqueNamespace is not null)
+                {
+                    writer.WriteString("defaultNamespace", uniqueNamespace.NamespaceName);
+                }
+                if (uniqueDatabase is not null)
+                {
+                    writer.WriteString("defaultDatabase", uniqueDatabase.DatabaseName);
+                }
+
+                writer.WriteStartObject("authentication");
+                writer.WriteString("protocol", "ws");
+                // How to do host resolution?
```

Comment:
The `EndpointReference` type has a `Host` property which should be the host of the allocated endpoint (since they are containers it'd be the container network reference IIRC). Or maybe you can use the container name, like https://github.com/dotnet/aspire/blob/d621f86daff51dcdbc8860858b9571ce64ecb44f/src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs#L548 does

Reply:
It has to be the host because it is calling the surreal dependency from the container but from the client. It works with `endpoint.Host`.